### PR TITLE
Create and delete one item

### PR DESCRIPTION
**Pull request by bwillett2003**
This pull request introduces several key updates and improvements. In the ItemsController, the create action now includes error handling for ActionController::ParameterMissing, ensuring a 422 error is returned when required parameters are missing. Similarly, in the destroy action, error handling for ActiveRecord::RecordNotFound has been added to return a 404 error when attempting to delete a non-existent item.

Comprehensive RSpec request specs have been added. These include tests for successfully creating and destroying items, as well as sad path tests for missing or invalid parameters. The tests cover scenarios such as missing unit_price or name and attempting to delete a non-existent item, ensuring better error handling and coverage for edge cases.

### CODE REVIEW

File: app/controllers/api/v1/items_controller.rb [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+  def error_messages(messages, status)
+    {
+      errors: messages.map do |message|
+        {
+          status: status,
+          message: message
+        }
+      end
+    }
+  end
```
Comment (SThompson05) [inline app/controllers/api/v1/items_controller.rb:41]: Great job on using map here. Overall well done on this code.
